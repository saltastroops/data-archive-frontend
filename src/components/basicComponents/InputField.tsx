import * as React from "react";
import styled from "styled-components";

interface IInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  error?: string;
}

/**
 * A paragraph with an error message.
 */
const Error = styled.p.attrs({
  className: "has-text-danger"
})``;

/**
 * An input field with an error message underneath it.
 *
 * The field accepts all attributes you can pass to an HTML input element. In
 * addition, you may pass an error argument with an error message to display.
 */
const InputField = (props: IInputProps) => {
  const { className, error, ...rest } = props;
  return (
    <div className="control">
      <input
        className={`input ${error && error !== "" ? "is-danger" : ""}`}
        {...rest}
      />
      {error && <Error>{error}</Error>}
    </div>
  );
};
export default InputField;
