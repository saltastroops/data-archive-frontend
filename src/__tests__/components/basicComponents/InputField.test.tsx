import { ReactWrapper, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import InputField from "../../../components/basicComponents/InputField";

describe("InputField", () => {
  it("should render", () => {
    expect(<InputField />).toBeDefined();
  });

  it("render correctly", () => {
    expect(toJson(shallow(<InputField />))).toMatchSnapshot();
    expect(
      toJson(shallow(<InputField error={"it has error"} />))
    ).toMatchSnapshot();
    expect(
      toJson(
        shallow(<InputField name="name" value="value" placeholder="holder" />)
      )
    ).toMatchSnapshot();
  });

  it("should call onChange", () => {
    const onChange = jest.fn();
    const inputField = shallow(<InputField onChange={onChange} />);
    const input = inputField.find("input");
    input
      .find("input")
      .simulate("change", { target: { value: "Input Value" } });
    expect(onChange).toHaveBeenCalled();
  });
});
