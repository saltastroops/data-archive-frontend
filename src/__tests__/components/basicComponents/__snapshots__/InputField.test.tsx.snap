// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField render correctly 1`] = `
<div
  className="control"
>
  <input
    className="input "
  />
</div>
`;

exports[`InputField render correctly 2`] = `
<div
  className="control"
>
  <input
    className="input is-danger"
  />
  <styled.p>
    it has error
  </styled.p>
</div>
`;

exports[`InputField render correctly 3`] = `
<div
  className="control"
>
  <input
    className="input "
    name="name"
    placeholder="holder"
    value="value"
  />
</div>
`;
